astropy/io/fits/hdu/compressed/header.py

```diff
@@ -648,7 +473,7 @@ def _image_header_to_bintable_header_and_coldefs(
 
         for k in to_remove:
             try:
-                del bintable_header[k]
+                del bintable.header[k]
```

Comment:
For a freshly created table, I think this whole `else` clause can go.

Reply:
Fixed